### DISCUSSION TITLE
halo2 的 recursive 过程是可以像nova那样在多个prover之间流传还是只针对一个prover的多个instances做batch处理？

### DISCUSSION THREAD

**Discussion by Demian101** [z2o-k7e/halo2-step-by-step#30] [Q&A]
@Walker 提问，如题

**Comment by Demian101** [comment]
@周洋

我们面临的 halo2 问题:
- zcash IPA : 支持 Recursive，主打的是后面做 folding schemes
- PSE: 把 IPA 换成 KZG10，回归到了 PLONK 生态比较好，能上生产，变成了超大号的 PLONK（UltraPLONK）。能否实现 Recursive ，这是一个问题，目前暂时没有 Recursive 的一些 feature.

**Comment by qizhou** [comment]
Halo2的recursive好像最近PSE没有太多动作，我看到几个主要在ark的库上面：

https://github.com/privacy-scaling-explorations/multifolding-poc
https://github.com/privacy-scaling-explorations/folding-schemes
